lib/src/visitors/expression_parser_visitor.dart

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/error/error.dart';
+import 'package:angular_ast/src/ast.dart';
+import 'package:angular_ast/src/exception_handler/exception_handler.dart';
+import 'package:angular_ast/src/visitor.dart';
+
+class ExpressionParserVisitor<C> implements TemplateAstVisitor<TemplateAst, C> {
```

Comment:
this can be TemplateAstVisitor<Null, C> and then your methods don't have to return (since the visitor does state modifications only)